src/FixtureBuilder/Denormalizer/Fixture/SimpleFixtureBagDenormalizer.php

```diff
@@ -69,22 +75,30 @@ public function denormalize(array $data): FixtureBag
                     )
                 );
             }
+            try {
+                foreach ($rawFixtureSet as $reference => $specs) {
+                    if (null === $specs) {
+                        $specs = [];
+                    }
 
-            foreach ($rawFixtureSet as $reference => $specs) {
-                if (null === $specs) {
-                    $specs = [];
+                    $fixtures = $this->fixtureDenormalizer->denormalize(
+                        $fixtures,
+                        $fqcn,
+                        $reference,
+                        $specs ?? [],
+                        $flags
+                    );
+                }
+            } catch (UnexpectedValueException $e) {
```

Comment:
just for consistency sake with the rest of the codebase I would keep `$exception` instead of `$e` here